### DISCUSSION TITLE
Adding access control policy

### DISCUSSION THREAD

**Discussion by XinboEddie** [developmentseed/eoAPI#196] [Ideas]
Hello folks,
I recently deployed eoAPI on AWS and find it hard to customize access roles. The endpoint will expose the entire catalog, collection, and item contents. The built in IAM authorizer under the API gateway is my current solution, and all I need to do is adding IAM policy to user groups. If there is a place to manage access, please let me know. Also, it might also be a good idea to mention this on the set up document.

**Comment by alukach** [comment]
Hi @XinboEddie. We're working towards expanding auth solutions with eoAPI and improving documentation around the various strategies one might take.  How did this work out for you?

I admit that I've never used the IAM authorizer. With this strategy, any request must be signed with AWS access keys, correct?

Do you have any need for finer grained controls (ie surfacing a subset of a services functionality to certain users)?

**Reply by XinboEddie** [reply]
Hey @alukach, our original idea was to have this as an internal developer tool for data scientists. The IAM authorizer will use AWS IAM roles to determine who can access processed geotiffs. However, we are currently moving away from this idea.

**Reply by alukach** [reply]
Interesting! I'm currently on a journey of trying to best understand the auth requirements for various users of the eoAPI. I'd be keen to jump on a call and discuss what you ended up doing if you have the interest/bandwidth. You can ping me at `anthony@developmentseed.org`

**Reply by alukach** [reply]
@XinboEddie It's 10 months later, but we've released https://github.com/developmentseed/stac-auth-proxy/ to support auth scenarios on STAC APIs. Reach out if you want to chat about running it on your project!

**Reply by louiscb** [reply]
Hi @alukach is the stac-auth-proxy designed to work across the entire EOapi stack? I.e can it be used for auth and access control for both the stac api and tipg?

**Reply by alukach** [reply]
> is the stac-auth-proxy designed to work across the entire EOapi stack? 

Not explicitly. It's pretty targeted towards STAC at the moment. It's possible that someone would be able to reapply it to other services but it may be a bit unconventional. I think there's more work to do regarding building a complete auth solution for the entirety of eoAPI.
